admin/class-yoast-form-select.php

```diff
@@ -0,0 +1,131 @@
+<?php
+/**
+ * @package WPSEO\Admin
+ */
+
+/**
+ * Class for generating a html select.
+ */
+class Yoast_Form_Select {
+
+	/**
+	 * @var string The field name for the select.
+	 */
+	private $field_name;
+
+	/**
+	 * @var string Complete name for the selects' name attribute.
+	 */
+	private $select_name;
+
+	/**
+	 * @var array Array with the options to parse.
+	 */
+	private $options;
+
+	/**
+	 * @var string The current selected option.
+	 */
+	private $selected_option;
+
+	/**
+	 * Constructor.
+	 *
+	 * @param string $field_name      The field name for the select.
+	 * @param string $select_name     Complete name for the selects' name attribute.
+	 * @param array  $options         Array with the options to parse.
+	 * @param string $selected_option The current selected option.
+	 */
+	public function __construct( $field_name, $select_name, array $options, $selected_option ) {
+		$this->field_name      = $field_name;
+		$this->select_name     = $select_name;
+		$this->options         = $options;
+		$this->selected_option = $selected_option;
+	}
+
+	/**
+	 * Returns the generated html for the select.
```

Comment:
What do we do when no options are provided?
We want to show the `select` because we expect the variable in the save, but if the options are generated conditionally  we can never unset a value if there are no options.